sql, ui: add SQL commenter support

cthumuluru-crdb opened the pull request:
To support SQL commenter integration, parser should also take comment mode as another option. Using functional options the code looks much cleaner.

cthumuluru-crdb committed:
sql: refactor parser.go using functional options

To support SQL commenter integration, parser should
also take comment mode as another option. Using functional
options the code looks much cleaner.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/141980)
<!-- Reviewable:end -->

cthumuluru-crdb committed:
sql: retain SQL commenter tags

SQL commenter library is used by ORM libraries and customer applications
to add application context to the queries using inline SQL comments.

This PR includes the following changes:
- Retains comments during parsing phase.
- If the comments follow [SQL commenter specification](https://google.github.io/sqlcommenter/spec/#introduction),
they are converted to SQL tags.
- Changes to insights stats to also include SQL commenter tags.

cthumuluru-crdb committed:
ui: show SQL commenter tags on Admin Console Insights page

cthumuluru-crdb commented:
I think putting SQL commenter behind a cluster setting would improve this PR further. That work is still pending.

cthumuluru-crdb committed:
fixup! sql: retain SQL commenter tags

cthumuluru-crdb commented:
Observability took over the work to integrate SQL commenter. This PR is no longer needed.

kyle-a-wong commented:
As per Chandra's comment, https://github.com/cockroachdb/cockroach/pull/145435 contains the sqlcommenter relateed changes
